### ISSUE TITLE
About training time changes

### ISSUE THREAD

**Issue by panchengl** [shibo-shen/Dynamic-Channel-Propagation#1]
hello， thanks your work， when I use dynamic pruning, will the training time be longer than the time without dynamic pruning?

**Comment by shibo-shen** [comment]
Thanks for your question, and my answer is yes. The training time will be longer since we need to mask the output channels (dropout) and update all channel utilities, in the feed-forward and feed-backward pass, respectively. However, since the calculations are highly paralleized by applying GPUs, the extra time overhead is acceptable. In fact, in our later research we also found that, the dynamic channel selection process could be ended early, as soon as the structure of the neural network is basically fixed. Hoping This Helps.

**Comment by panchengl** [comment]
Thank you very much for your reply, I will try to reproduce your work in the near future, thank you again for your work
